x.py

```diff
@@ -116,14 +116,37 @@ def build(dir: str, jobs: int, ghproxy: bool, ninja: bool, unittest: bool, compi
         target.append("unittest")
     run(cmake, "--build", ".", f"-j{jobs}", "-t", *target, verbose=True, cwd=dir)
 
-def cpplint() -> None:
-    command = find_command("cpplint", msg="cpplint is required")
-    options = ["--linelength=120", "--filter=-build/include_subdir,-legal/copyright,-build/c++11"]
-    sources = [*glob("src/*.h"), *glob("src/*.cc")]
-    run(command, *options, *sources, verbose=True)
+def get_source_files() -> List[str]:
+    return [
+        *glob("src/**/*.h", recursive=True),
+        *glob("src/**/*.cc", recursive=True),
+        *glob("tests/cppunit/**/*.h", recursive=True),
+        *glob("tests/cppunit/**/*.cc", recursive=True),
+        *glob("tools/kvrocks2redis/**/*.h", recursive=True),
+        *glob("tools/kvrocks2redis/**/*.cc", recursive=True),
+    ]
+
+def check_format(clang_format_path: str, i: bool) -> None:
+    print("WARNING: We use clang-format 12 in CI,\n"
+        "  so we recommend that you also use this version locally to avoid inconsistencies.\n"
+        "  You can install it from your package manager (usually in clang-12 package)\n"
+        "  or download it from https://github.com/llvm/llvm-project/releases/tag/llvmorg-12.0.1")
```

Comment:
Shall we always show warning?

Reply:
now I will only appear when clang-format in not versioned 12